src/api.js

```diff
@@ -1,46 +1,70 @@
-const rest = require(`./rest`);
-const ws = require(`./ws`);
-let type;
-
-/**
- * Initialize API based on protocol and URL
- * 
- * @param {String} serverURL 
- * @returns 
- */
-function init(serverURL){
-  type = serverURL.slice(0, 2);
-  if (type === `ht`){
-    return rest.init(serverURL);
-  } else {
-    return ws.init(serverURL);
-  }
-}
+'use strict';
 
-/**
- * Returns api type
- * @returns {*}
- */
-function getType(){
-  return type;
-}
+// Requirements
+
+const Rest = require(`./rest`);
+const WS = require(`./ws`);
+
+
+// Source
+
+class API {
+    /**
+     * @param {Object} options { authServiceURL, mainServiceURL, pluginServiceURL }
+     */
+    constructor(options) {
+        const { authServiceURL, mainServiceURL, pluginServiceURL } = options;
+
+        const SelectedAPI = this.getType(mainServiceURL);
 
-/**
- * Returns current API
- * 
- * @returns 
- */
-function getAPI(){
-  if (type === `ht`){
-    return rest;
-  } else {
-    return ws;
-  }
+        if (SelectedAPI) {
+            this.schema = new SelectedAPI({
+                authServiceURL,
+                mainServiceURL,
+                pluginServiceURL
+            });
+        } else {
+            throw new Error('unexpected mainServiceURL, please use allowed protocol');
+        }
+
+        return this.schema;
```

Comment:
Constructor shouldn't return any value. It is better to leave the default behavior when constructor returns itself by itself. Other behavior can confuse a bit.

Reply:
OK, it was a hotfix according to current code.